http/src/request/channel/create_pin.rs

```diff
@@ -51,4 +53,13 @@ impl<'a> CreatePin<'a> {
     }
 }
 
+impl<'a> AuditLogReason for CreatePin<'a> {
```

Comment:
Some static assertions on all these requests to assert that the AuditLogReason is implemented for them would be nice